os/kernel/binary_manager/binary_manager_bootparam.c

```diff
@@ -0,0 +1,147 @@
+ *
+ ****************************************************************************/
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+#include <tinyara/config.h>
+#include <debug.h>
+#include <stdio.h>
+#include <string.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <stdbool.h>
+
+#include <tinyara/binary_manager.h>
+
+#include "binary_manager.h"
+
+/* Data for Boot parameters */
+static binmgr_bpinfo_t bp_info;
+
+#define BP_SEEK_OFFSET(index)   (index == 0 ? 0 : BOOTPARAM_PARTSIZE / 2)
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+/****************************************************************************
+ * Name: binary_manager_register_bppart
+ *
+ * Description:
+ *	 This function registers the partition of boot parameters.
+ *
+ ****************************************************************************/
+void binary_manager_register_bppart(int part_num, int part_size)
+{
+	if (part_num < 0 || part_size != BOOTPARAM_PARTSIZE) {
+		bmdbg("ERROR: Invalid part info : num %d, size %d\n", part_num, part_size);
+		return;
+	}
+
+	bp_info.part_num = part_num;
+
+	bmvdbg("[BOOTPARAM] part num %d\n", part_num);
+}
+
+bool is_valid_bootparam(binmgr_bpdata_t *bp_data)
+{
+	if (!bp_data) {
+		bmdbg("Boot param is NULL\n");
+		return false;
+	} else if (bp_data->format_ver == 0 || bp_data->active_idx >= binary_manager_get_kdata()->part_count) {
+		bmdbg("Invalid data. ver: %u, active index: %u, addresses: %x, %x\n", bp_data->version, bp_data->active_idx, bp_data->address[0], bp_data->address[1]);
+		return false;
+	} else if (bp_data->crc_hash != crc32((uint8_t *)bp_data + CHECKSUM_SIZE, sizeof(binmgr_bpdata_t) - CHECKSUM_SIZE)) {
+		bmdbg("Invalid crc32 value, crc32 %u, ver: %u, active index: %u, addresses: %x, %x\n", bp_data->crc_hash, bp_data->version, bp_data->active_idx, bp_data->address[0], bp_data->address[1]);
+		return false;
+	}
+
+	return true;
+}
+
+/*****************************************************************************************************
+ * Name: binary_manager_scan_bootparam
+ *
+ * Description:
+ *	 This function updates the latest boot parameter information by scanning all boot parameters.
+ *
+ *****************************************************************************************************/
+int binary_manager_scan_bootparam(void)
+{
+	int fd;
+	int ret;
+	int bp_idx;
+	int latest_idx = -1;
+	uint32_t latest_ver = 0;
+	binmgr_bpdata_t bp_data[BOOTPARAM_COUNT];
+	char bp_devpath[BINARY_PATH_LEN];
+
+	/* Open dev to access boot parameters */
+	snprintf(bp_devpath, BINARY_PATH_LEN, BINMGR_DEVNAME_FMT, bp_info.part_num);
+	fd = open(bp_devpath, O_RDWR, 0666);
+	if (fd < 0) {
+		bmdbg("ERROR: Get a fd of bootparam, %d\n", fd);
```

Comment:
How about printing the `errno` instead of `fd`?
Because `open` returns just `ERROR` when fail-case, but it is not helpful.